sphinx/source/Concepts.rst

```diff
@@ -132,6 +139,21 @@ This address is not permanent, and may change at any time if address claiming oc
 Now that you understand these concepts, you can create your first control function and send some basic messages!
 Check out the :doc:`basic tutorial <Tutorials/The ISOBUS Hello World>` now if you'd like to practice this, or keep reading to learn more.
 
+.. image:: ../images/concepts_2.png
+   :width: 241px
+   :align: right
+   :alt: Minimal Block Diagram of an example ECU
+
+An example use of **ArIsoStack++** might be to create an ISOBUS ECU that controls an implement, like a seed drill or sprayer. This ECU would implement a Hardware Interface with CAN, then set up a number of Control Functions: internal Control Function's to receive commands and some external Control Function's for commanding with other modules. 
+
+The ECU would implement the Virtual Terminal(VT) client to send an Object Pool (IOP) file to a VT to display controls and information from the "screen in the tractor cab", and a Task Controller (TC) client to receive a tasking job. The rest of the Application would be the logic to control the implement based on the TC job and commands from the VT.
```

Comment:
Also here a bit too much into detail into actual implementation where we are only diving into the concepts really. I think a good place to put this is (almost) at the bottom of the [API Documentation](https://agisostack-plus-plus.readthedocs.io/en/latest/api/index.html), just before the blue note. Then everyone can see what an example would be and what documentation to open. 

Another place would be in some tutorial, but I don't think we have a single tutorial that fits this diagram well, hmm...

Reply:
I'll think about another tutorial in the future as I get my feet wet on the library a little more - an ECU stack example would be pretty sweet.